src/snorkel_extractor_memex/dataset_utils.py

```diff
@@ -146,11 +146,10 @@ def replace_middle_double_quotes(text):
     
     string text: string to modify
     """
-    indices = [m.start(0) for m in re.finditer(r'[a-zA-Z0-9_.!?]"[a-zA-Z0-9_.!?]', text)]
-    indices = indices + [m.start(0) for m in re.finditer(r'[a-zA-Z0-9_.!?]"\s[a-zA-Z0-9_.!?]', text)]
-    indices = indices + [m.start(0) for m in re.finditer(r'[a-zA-Z0-9_.!?]\s"[a-zA-Z0-9_.!?]', text)]
+    indices = [m.start(0) for m in re.finditer(r'[^:,][^{]"[^:,}]', text)]
```

Comment:
Could you add a comment to describe exactly what this is doing?  What is this change doing?